aiida_common_workflows/workflows/eos.py

```diff
@@ -0,0 +1,158 @@
+            help='The number of points to compute for the equation of state.')
+        spec.input('scale_increment', valid_type=orm.Float, default=lambda: orm.Float(0.02),
+            validator=validate_scale_increment,
+            help='The relative difference between consecutive scaling factors.')
+        spec.input('generator_inputs', valid_type=orm.Dict, validator=validate_generator_inputs,
+            help='The inputs that will be passed to the inputs generator of the specified `sub_process`. It should '
+            'include all the inputs, excluding the `structure` and `previous_workchain`.')
+        spec.input_namespace('sub_process', dynamic=True, populate_defaults=False)
+        spec.input('sub_process_class', non_db=True, validator=validate_sub_process_class)
+        spec.inputs.validator = validate_inputs
+        spec.outline(
+            cls.run_init,
+            cls.run_eos,
+            cls.inspect_eos,
+        )
+        spec.output_namespace('structures', valid_type=orm.StructureData,
+            help='The relaxed structures at each scaling factor.')
+        spec.output_namespace('total_energies', valid_type=orm.Float,
+            help='The computed total energy of the relaxed structure at each scaling factor.')
+        spec.exit_code(400, 'ERROR_SUB_PROCESS_FAILED',
+            message='At least one of the `{cls}` sub processes did not finish successfully.')
+
+    def get_scale_factors(self):
+        """Return the list of scale factors."""
+        if 'scale_factors' in self.inputs:
+            return self.inputs.scale_factors
+
+        count = self.inputs.scale_count.value
+        increment = self.inputs.scale_increment.value
+        return [orm.Float(1 + i * increment - (count - 1) * increment / 2) for i in range(count)]
+
+    def get_sub_workchain_builder(self, scale_factor, previous_workchain=None):
+        """Return the builder for the relax workchain."""
+        structure = scale_structure(self.inputs.structure, scale_factor)
+        process_class = WorkflowFactory(self.inputs.sub_process_class)
+        generator_inputs = self.inputs.generator_inputs.get_dict()
+
+        builder = process_class.get_inputs_generator().get_builder(
+            structure,
+            calc_engines=generator_inputs['calc_engines'],
+            protocol=generator_inputs['protocol'],
+            relaxation_type=RelaxType(generator_inputs['relaxation_type']),
+            threshold_forces=generator_inputs.get('threshold_forces', None),
+            threshold_stress=generator_inputs.get('threshold_stress', None),
+            previous_workchain=previous_workchain
+        )
+        builder._update(**self.inputs.get('sub_process', {}))  # pylint: disable=protected-access
+
+        return builder
+
+    def run_init(self):
+        """Run the first workchain."""
+        scale_factor = self.get_scale_factors()[0]
```

Comment:
First of all, I would like to get confirmation from @broeder-j that the  'lowest volume' f he mentioned in the comments it is actually the "smallest volume" or if he meant something else. Then, here we assume that the first scaling factor in the list is the smallest one. Not necessary the case for manually passed scaling factors. I would implement a sorting of the list in `get_scale_factors`

Reply:
I don't think it should matter which scaling factor we choose to be the reference, does it?

Reply:
This is why I'm interested to hear @broeder-j opinion here. For siesta for sure it does not matter